### ISSUE TITLE
Websocket connections not properly closed when switching regions

### ISSUE THREAD

**Issue by arein** [WalletConnect/web-examples#65]
Expected behavior:
when I switch regions the existing websocket is closed.

Actual behavior:
It stays open

See https://www.loom.com/share/8dada9ce011f4e1d8dca42d499370feb

**Comment by ganchoradkov** [comment]
> @ganchoradkov in your experimentation with manual websocket disconnecting at the client level, did you find a reliable way to do this?
> 
> Or is this a matter of implementing something like `transportClose` first as we discussed here: https://walletconnect.slack.com/archives/C03RME3BS9L/p1663859970679629

@bkrem, the way I managed to handle disconnect is more of a work-around and we really need to implement something like  `transportClose`

Currently, the only way to disconnect the socket is to call `core.relayer.provider.connection.close()` and to disable 2 event loops

- heartBeat - it does jsonRPC request which reconnects the socket
- `RELAYER_PROVIDER_EVENTS.disconnect` which listens for socket disconnect & reconnects it after a given delay

changing the relay url is only possible during `core` initialization.

The bottom line is - we need the `transportClose` to handle the above steps automatically

**Comment by bkrem** [comment]
@ganchoradkov added an issue here (https://github.com/WalletConnect/walletconnect-monorepo/issues/1507)

Added all the relevant use cases I was aware of, please correct/augment as necessary based on your findings :pray:

**Comment by bkrem** [comment]
Superseded by #149
